src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecs.jl

```diff
@@ -0,0 +1,96 @@
+"""
+StateVecs
+
+  Provides a state vector for a set of variables.
+
+# Interface
+ - [`StateVec`](@ref) contains a vector of variables for every
+                      mesh point, and a name mapping NamedTuple
+ - [`over_sub_domains`](@ref) return a range over subdomains
+ - [`Slice`](@ref) index slice along dimension
+"""
+module StateVecs
+
+import Base
+using Test, Pkg, ..Grids
+export StateVec, over_sub_domains, Slice
+
+struct FieldsPerElement{T}
+  vals::Vector{T}
+end
+
+function Base.show(io::IO, fpe::FieldsPerElement)
+  print("fpe = ",fpe.vals)
+end
+
+struct StateVec{T, I, NT}
```

Comment:
Worth seeing if we can use [StructArrays](https://github.com/piever/StructArrays.jl) for this

Reply:
I remember now why I avoided `StructArray`. The memory layout is restricted by the fact that members are accessed by `struct_array.x` (every member is now a 2D matrix containing a value for every grid element and subdomain).

Using StructArrays or not is not a huge issue since the components I'm adding are only to produce an MWE (and doesn't need to be performant), but I wanted to fully encapsulate the memory layout so that it could be changed if necessary.

That said, I do like the way members are accessed with the `.x` rather than `[:x]` as I currently do.

Reply:
why is that restriction an issue? (I'm just curious here)